### PR TITLE
[WORKAROUND][COMGR] W/A for issue 1359: Use hipModuleLoad with interim file instead of hipModuleLoadData. By-product: Better names for tempfiles.

### PR DESCRIPTION
**Pull request by atamazov**
Quick W/A for issue #1359, almost identical to https://github.com/AMDComputeLibraries/MLOpen/pull/2501.

### CODE REVIEW

File: src/temp_file.cpp [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #include <boost/filesystem.hpp>
 
 namespace miopen {
-TempFile::TempFile(const std::string& path_template) : name(path_template), dir("tmp")
+TempFile::TempFile(const std::string& path_infix) : name("file"), dir(path_infix)
```
Comment (atamazov) [inline src/temp_file.cpp:7]: This changes directory name from "`miopen-tmp-...`" to "`miopen-<INFIX>-...`" which is more informative for the geeks who would like to inspect the content of the system temp directory.

**Comment by atamazov** [comment]
CI passed, please review

File: src/include/miopen/temp_file.hpp [hunk 2]
```diff
@@ -36,23 +36,26 @@ namespace miopen {
...
+        : path_infix(std::move(other.path_infix)), dir(std::move(other.dir))
+    {
+    }
 
     TempFile& operator=(TempFile&& other) noexcept
     {
-        name = std::move(other.name);
-        dir  = std::move(other.dir);
+        path_infix = std::move(other.path_infix);
+        dir        = std::move(other.dir);
```
Comment (pfultz2) [inline src/include/miopen/temp_file.hpp:49]: Why is a custom assignment and move constructor necessary as there is no custom destructor in this class? Can't this be written as:

```cpp
TempFile(TempFile&& other) noexcept = default;
TempFile& operator=(TempFile&& other) noexcept = default;
```

Comment (atamazov) [reply src/include/miopen/temp_file.hpp:49]: @pfultz2 Hmm... The default should work in this case.

@DrizztDoUrden This first appeared at https://github.com/ROCmSoftwarePlatform/MIOpen/pull/280/files#r775664957. Any special considerations there?

Comment (atamazov) [reply src/include/miopen/temp_file.hpp]: @pfultz2 Resolved

Comment (DrizztDoUrden) [reply src/include/miopen/temp_file.hpp]: No, nothing special, I think.

**Review by DrizztDoUrden — APPROVED** [review]
LGTM

**Pull request merged**